From 183cdabdd884fec047676dffa6dfb823c8bfe02b Mon Sep 17 00:00:00 2001
From: Changhyeok Bae <changhyeok.bae@lge.com>
Date: Wed, 8 Jun 2016 11:18:19 +0900
Subject: [PATCH] Fix -reduce-relocations error

error: #error "You must build your code with position independent code
       if Qt was built with -reduce-relocations. " "Compile your code
       with -fPIC (-fPIE is not enough)."
|  #  error "You must build your code with position independent code
   if Qt was built with -reduce-relocations. "
---
 src/hmi/qml/hmi-launcher/CMakeLists.txt |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/hmi/qml/hmi-launcher/CMakeLists.txt b/src/hmi/qml/hmi-launcher/CMakeLists.txt
index ab6b33a..774a377 100644
--- a/src/hmi/qml/hmi-launcher/CMakeLists.txt
+++ b/src/hmi/qml/hmi-launcher/CMakeLists.txt
@@ -69,6 +69,8 @@ set(CMAKE_AUTOMOC ON)
 add_definitions(${QT5_CORE_DEFINITIONS})
 add_definitions(-DQT_PLUGIN)
 add_definitions(-DQT_NO_DEBUG)
+# FIX: Qt was built with -reduce-relocations
+add_definitions(-fPIC)
 set(qt_moc ${QT_MOC} -o)
 set(MOC moc)
 set(MOC_DIR ${CMAKE_CURRENT_BINARY_DIR}/${MOC})
-- 
1.7.9.5

